### ISSUE TITLE
When is the cvae_model saved in Citation?

### ISSUE THREAD

**Issue by Xiuchen519** [SongtaoLiu0823/LAGNN#1]
Hi, author! Thank you for opening the source code!
According to the description of  ’Active Learning Trick‘ in the appendix, for Citation dataset, I think `cvae_model` should be saved after calling `generated_generator` in `cvae_pretrain.py`. But I can't find the code to save  it.  And I find another file `cvae_tune.py`, it  doesn't seem to work, since there is no `generated_features` function in the `cvae_pretrain.py`.  I just feel uncertain about that when `cvae_model` should be saved for Citation dataset. Hope to get your reply! Thank you!

**Comment by SongtaoLiu0823** [comment]
> Hi, author! Thank you for opening the source code!
> According to the description of ’Active Learning Trick‘ in the appendix, for Citation dataset, I think `cvae_model` should be saved after calling `generated_generator` in `cvae_pretrain.py`. But I can't find the code to save it. And I find another file `cvae_tune.py`, it doesn't seem to work, since there is no `generated_features` function in the `cvae_pretrain.py`. I just feel uncertain about that when `cvae_model` should be saved for Citation dataset. Hope to get your reply! Thank you!

Hi,

We use the `cvae_tune.py` to tune the hyper-parameters of CVAE for citation dataset. After tuning, you can use `cvae_pretrain.py` to get the generator and save it.

**Comment by SongtaoLiu0823** [comment]
Hi,

Apologies for the previous misunderstanding. I realized that I had not updated the function name when submitting my code to the GitHub repository. The correct function name should be "generated_generator" instead of "generated_features." I have now fixed this issue.

For each hyperparameter set, I use cvae_pretrain.py to generate features, and then use cvae_tune.py to evaluate the CVAE model using these generated features. I select the hyperparameter set with the highest accuracy on the validation set.

Indeed, in comparison to the OGB dataset, I had not previously written cvae_generate_citation.py for citation datasets. However, I have now created this script, which can be found here: [cvae_generate_citation.py](https://github.com/SongtaoLiu0823/LAGNN/blob/main/Citation/cvae_generate_citation.py).

Thank you!
